tracer/test/Datadog.Trace.Security.IntegrationTests/IAST/AspNetCore2IastTests.cs

```diff
@@ -116,6 +116,43 @@ await VerifyHelper.VerifySpans(spans, settings)
                           .UseFileName(filename)
                           .DisableRequireUniquePrefix();
     }
+
+    // When the request is finished without the header Strict-Transport-Security or with an invalid value on it, we should detect the vulnerability and send it to the agent when these conditions happens:
+    // The connection protocol is https or the request header X-Forwarded-Proto is https
+    // The Content-Type header of the response looks like html(text/html, application/xhtml+xml)
+    // Header has a valid value when it starts with max-age followed by a positive number (>0), it can finish there or continue with a semicolon ; and more content.
+
+    [SkippableTheory]
+    [Trait("Category", "ArmUnsupported")]
+    [Trait("RunOnWindows", "True")]
+    [InlineData("text/html;charset=UTF-8", 200, "max-age=31536000", "https")]
+    [InlineData("application/xhtml%2Bxml", 200, "max-age%3D10%3Botherthings", "https")]
+    [InlineData("text/html", 500, "invalid", "https")]
+    [InlineData("text/html", 200, "invalid", "")]
+    [InlineData("text/plain", 200, "invalid", "https")]
+    [InlineData("text/html", 200, "", "https")]
+    [InlineData("application/xhtml%2Bxml", 200, "", "https")]
+    [InlineData("text/html", 200, "invalid", "https")]
+    public async Task TestStrictTransportSecurityHeaderMissing(string contentType, int returnCode, string hstsHeaderValue, string xForwardedProto)
+    {
+        var queryParams = "?contentType=" + contentType + "&returnCode=" + returnCode +
+            (string.IsNullOrEmpty(hstsHeaderValue) ? string.Empty : "&hstsHeaderValue=" + hstsHeaderValue) +
+            (string.IsNullOrEmpty(xForwardedProto) ? string.Empty : "&xForwardedProto=" + xForwardedProto);
+        var filename = "Iast.StrictTransportSecurity.AspNetCore2." + contentType.Replace("/", string.Empty) +
+            "." + returnCode.ToString() + "." + (string.IsNullOrEmpty(hstsHeaderValue) ? "empty" : hstsHeaderValue)
+            + (string.IsNullOrEmpty(xForwardedProto) ? "empty" : xForwardedProto);
```

Comment:
Up to you if you want to include this, but you're missing a `.` in the middle of the last two parameters for these AspNetCore2 snapshot file names.

```suggestion
            + "." + (string.IsNullOrEmpty(xForwardedProto) ? "empty" : xForwardedProto);
```